### PR TITLE
Sadashay branch

### PR DESCRIPTION
**Pull request by SadashayKanungo**
# Changes
## Added Final changes in **Collection Class** ( created in #14 ).
This class implements the Generate functionality for generating NFT Image and Metadata files from Layer images.
## Created **FileStorage Abstract Class** and the following classes implementing it.
These classes have functions for uploading NFT Collections and Single NFTs to a File Storage platform.
- **NFT.storage Class** for uploading to [NFT.storage](https://nft.storage/)
- **Pinata Class** for uploading to [Pinata](https://www.pinata.cloud/)
- **Storj Class** for uploading to [Storj](https://landing.storj.io/permanently-pin-with-storj-dcs)
- **Infura Class** for uploading to [Infura](https://infura.io/product/ipfs)
- **Arweave Class** for uploading to [Arweave](https://www.arweave.org/) using [Bundlr](https://bundlr.network/)
## Created **Contract Class**.
This class implements the following functionalities.
- Programatically generating (_"Drafting"_) ERC721 and ERC1155 Smart contracts as solidity files.
- Deploying Smart contracts to any of the following Blockchain networks
    - [Ethereum](https://ethereum.org/): Mainnet, Ropsten, Rinkeby, Goerli, Kovan
    - [Polygon](https://polygon.technology/): Mainnet, Mumbai
- Interacting with deployed Smart contracts on any of the above networks
## Created Documentation for the project

### CODE REVIEW

**Comment by SadashayKanungo** [comment]
Please merge #25 before merging this.

**Comment by mbcse** [comment]
Add a credentials file for tests and add mention it in readme

**Comment by mbcse** [comment]
Lint the package !

**Comment by mbcse** [comment]
Add content type for nft asset while storing in storage services, you can use mime package

File: src/index.ts [hunk 1]
```diff
@@ -18,6 +29,141 @@ class Toolbox {
...
+				throw new Error("Unknown IPFS Service");
+		}
+	}
+
+	async uploadCollectionNFT() {
+		if (!this.collection) {
+			throw new Error("No Collection is initialized");
+		}
+		if (!this.fileStorageService) {
+			throw new Error("No IPFS Service is initialized");
```
Comment (mbcse) [inline src/index.ts:115]: Please change No IPFS to No Storage

**Comment by mbcse** [comment]
Add contributors guideline, docs and diagram

**Pull request merged**